### ISSUE TITLE
Error importing library 

### ISSUE THREAD

**Issue by kholsap12** [rmcrackan/Libation#1087]
Its not importing all books, only some from my library

[_Log202412_001.log](https://github.com/user-attachments/files/18211334/_Log202412_001.log)

**Comment by rmcrackan** [comment]
Sorry about this. Please try again to attach your log to this ticket. Something must have gone wrong last time.

**Comment by Ofahn** [comment]
[_Log202412.zip](https://github.com/user-attachments/files/18270688/_Log202412.zip)

I'm having the same problem.

**Comment by rmcrackan** [comment]
@Ofahn Thanks for including the log. Does the problem still happen with the latest pre-release? https://github.com/rmcrackan/Libation/releases/tag/v11.6.3

**Comment by Ofahn** [comment]
Thanks for getting back so quickly. I'll install and respond. I don't use PayPal.  Do you have another way of receiving funds? Sent from my Galaxy
-------- Original message --------From: rmcrackan ***@***.***> Date: 12/29/24  2:53 PM  (GMT-05:00) To: rmcrackan/Libation ***@***.***> Cc: Ofahn ***@***.***>, Mention ***@***.***> Subject: Re: [rmcrackan/Libation] Error importing library  (Issue #1087) 
@Ofahn Thanks for including the log. Does the problem still happen with the latest pre-release? https://github.com/rmcrackan/Libation/releases/tag/v11.6.3

**Comment by rmcrackan** [comment]
Sure, Thanks! You can buy amazon gift cards for any amount and send them to my email: rmcrackan@gmail.com

They suggest big amounts but you can put in pretty much anything into the "Enter amount" box.

https://www.amazon.com/gp/product/B07PCMWTSG

**Comment by Ofahn** [comment]
Downloaded 11.6.3.  Issue persists. The program isn’t recognizing some of the books in my Audible library.

 

Card is on the way.  Thanks for a GREAT program.

 

Eric

**Comment by rmcrackan** [comment]
@Ofahn Thanks for the gift card!!

Can you attach your latest log? I want to see if that pre-release is showing the same error or a different one.

**Comment by rmcrackan** [comment]
@Ofahn - The log didn't come through as an email attachment. You may have to drag/drop directly on the website.

I'm going to delete or edit your replies above since they include stuff from your email that should only be for you. I'm worried anyone could just click those links and unsubscribe you from this thread.

**Comment by Ofahn** [comment]
[_Log202412.zip](https://github.com/user-attachments/files/18271381/_Log202412.zip)

**Comment by rmcrackan** [comment]
Thank you for today's log. This shows no errors and also shows Version 11.1. Can you run the 11.6.3 version again? In the mean time, please check Libation > Settings > Trash Bin to see if the missing books are in there.

**Comment by Ofahn** [comment]
[_Log202412_001.zip](https://github.com/user-attachments/files/18271848/_Log202412_001.zip)

**Comment by rmcrackan** [comment]
Thanks. No errors here either; it was very helpful though for the 2nd bullet below. I have a few things to check:

* See if the title/s was/were accidentally removed. Look in Libation > Settings > Trash Bin
* There currently seems to be only 1 account getting scanned. It has ~125 books. However you have over 6,000 books in Libation. Is there another account which needs to be added? Look in Libation > Settings > Accounts

**Comment by Ofahn** [comment]
I also count 125 books in the library.  There’s about 10 or so which won’t download, or some of the downloaded filed continue to corrupt.

 

The large difference in numbers is due to the podcasts I downloaded.  Each episode appears to count as a book.

 

None of the problematic books are in the Trash Bin.

 

Does this help?

**Comment by rmcrackan** [comment]
Using the official app, are you able to download and play these problematic titles? Are they purchased, or from the Plus catalog?

**Comment by Ofahn** [comment]
I can play them on the Audible App.  Two have been purchased, the rest are from the Plus Catalog.

**Comment by rmcrackan** [comment]
Are their any filters at the top of Libation? I can't imagine why a purchased book wouldn't be included in your scan. Which titles are not getting to your grid?

**Comment by Ofahn** [comment]
No filters.

Ghostriders 1968-1975: "Mors De Caelis" Combat History of the AC-130 Spectre Gunship, Vietnam, Laos, Cambodia (Ghostriders, 1) is a book I purchased.  The program won't recognize it.  

I also purchased Disney's Land: Walt Disney and the Invention of the Amusement Park That Changed the World. Half the chaptered files downloaded in a corrupt form, and I've repeated that download several times with the exact same results.

The others are Plus books.

**Comment by rmcrackan** [comment]
For the corrupted files, does it help if you change the audio quality? Libation > Settings > Audio File Options tab, top left setting.

**Comment by Ofahn** [comment]
No, and it's bizarre.  Some of the chapters are usable, and some are corrupted.

**Comment by rmcrackan** [comment]
Are all chapters un-corrupted in the official app?

Regarding the other that Libation won't recognize: do you mean that the scan won't put it into the main grid?

**Comment by Ofahn** [comment]
The chapters play in the app.

The scan won't place the book in the grid.  I've manually scanned and auto-scanned several times.  No joy.

**Comment by rmcrackan** [comment]
I have no idea about the corrupt files :( I'm not sure I've encountered this before and I can't reproduce it.

Regarding that missing one with the long title, let's see if it's actually hiding in the database. Settings > Hangover > Database tab.
`select Title from Books where Title like '%ghost%'` then click Execute.

Did this find it? If yes, does it show up in Libation if you filter the main grid with `ghost`?

If that Hangover step didn't work, how about we back up your current database, get better logging, and start fresh to see if we can pinpoint the issue. This is a bit more involved:

* Libation > Settings > Important Settings tab, bottom left. Change logging level to debug. (After we've debugged this, you'll want to change it back to information. 'Debug' writes ridiculous amounts to the log.)
* In the same folder as your log files is your database: `LibationContext.db`. Back this up and keep it safe! Then delete LibationContext.db.
* Do a new scan. It should pick up the 125 titles fresh and hopefully log whether or not it's importing the 'Ghostriders 1968-1975' book.
* Send me the new log file

Whenever we're finished debugging, change log level back to information and restore the orig LibationContext.db.

**Comment by Ofahn** [comment]
I found the Hangover Tab, and then selected Database.  There's nothing in the SQL window.

Can you clarify this?

    select Title from Books where Title like '%ghost%' then click Execute.
    Did this find it? If yes, does it show up in Libation if you filter the main grid with ghost?

**Comment by rmcrackan** [comment]
In the Database tab of Hangover, there's a textbox in the top half. Paste this into it then click the execute button: `select Title from Books where Title like '%ghost%'` What does it say in the bottom half after clicking execute?

**Comment by Ofahn** [comment]
Ghostriders 1968-1975	
Ghostriders 1976-1995	
Ghostbuster's Daughter	
The Ghost Insurgency	
Spa-a-a-a-a-a-a-a-a-a-ce Ghost on Blu-ray	

5 results

**Comment by rmcrackan** [comment]
So that book is in the database but for some reason it's not displaying in your grid. The only reason I can think of is it's in the trash bin. If you're sure that's not the case, can you put this into Libation's filter bar and hit enter: `Ghostriders` . Does 'Ghostriders 1968-1975' appear in the grid?

**Comment by Ofahn** [comment]
Well; good news, bad news.

The book appeared in the grid and downloaded.  The files weren't corrupted.  YEAH.

Within the past few days each time I open 11.6.3 the program begins downloading files I haven't chosen.  See attached.
![Doc1-1](https://github.com/user-attachments/assets/b95c88c6-a70c-4648-8400-8156bd23c789)

**Comment by rmcrackan** [comment]
That sounds like the 'auto download' feature might be turned on. You can disable it under Settings > Import library tab, bottom checkbox.

**Comment by Ofahn** [comment]
Fixed, thanks.

**Comment by Ofahn** [comment]
Thanks.
